__tests__/app.test.js

```diff
@@ -159,8 +159,42 @@ describe("/api/articles/:article_id/comments", () => {
     return request(app)
       .get("/api/articles/banana/comments")
       .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("invalid id type");
+      });
+  });
+  test("POST: 201 insert a new comment to the given article id and responds with an object of the posted comment", () => {
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send({ username: "rogersop", body: "This article was brilliant." })
+      .expect(201)
+      .then(({ body }) => {
+        const { comment } = body;
+        expect(comment.hasOwnProperty("body")).toBe(true);
+        expect(comment.hasOwnProperty("votes")).toBe(true);
+        expect(comment.hasOwnProperty("author")).toBe(true);
+        expect(comment.hasOwnProperty("article_id")).toBe(true);
+        expect(comment.hasOwnProperty("created_at")).toBe(true);
+
+        expect(comment.username === "rogersop");
+        expect(comment.body === "This article was brilliant.");
+      });
+  });
+  test("POST: 400 responds with an error message when given a malformed body", () => {
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send({ username: "rogersop" })
+      .expect(400)
       .then(({ body }) => {
         expect(body.msg).toBe("bad request");
       });
   });
+  test("POST: 400 responds with an error message when given an invalid id", () => {
+    return request(app)
+      .post("/api/articles/banana/comments")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body.msg).toBe("invalid id type");
+      });
+  });
```

Comment:
Please add the following tests:
- Non-existent ID
- Username doesn't exist (404)